// @m3o/m3o-node/lib/stream.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Stream = void 0;
class Stream {
    constructor(conn, service, endpoint) {
        this.conn = conn;
        this.service = service;
        this.endpoint = endpoint;
    }
    send(msg) {
        return new Promise((resolve, reject) => {
            this.conn.send(msg);
        });
    }
    // this probably should use observables or something more modern
    recv(cb) {
        this.conn.on('message', (m) => {
            cb(JSON.parse(m));
        });
    }
}
exports.Stream = Stream;
// function marshalRequest(service: string, endpoint: string, v: any): string {
//   const jsonBody = JSON.stringify(v);
//   return JSON.stringify({
//     service: service,
//     endpoint: endpoint,
//     body: Buffer.from(jsonBody).toString('base64'),
//   });
// }
// function unmarshalResponse(msg: string): any {
//   const rsp: ClientResponse = JSON.parse(msg);
//   return Buffer.from(rsp.body, 'base64').toString();
// }
